File: app/src/main/assets/videos/shared/TikTok.go
package shared

import (
	"fmt"
	"net/http"
	"net/url"
	"strings"
)

func TikTok(uri string, proxy *url.URL) ([]byte, error) {
	// https://www.tiktok.com/@sonyakisa8/video/7156942763492166913?is_from_webapp=1&sender_device=pc&web_id=7146322967022372354
	uri = fmt.Sprintf("https://www.tikwm.com/api/?count=12&cursor=0&web=1&hd=1&url=%s", uri)
	result, err := Fetch(uri, nil, proxy, func(r *http.Request) {
		r.Header.Set("Accept", "application/json, text/javascript, */*; q=0.01")
		r.Header.Set("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
		r.Header.Set("sec-ch-ua", "\"Chromium\";v=\"104\", \" Not A;Brand\";v=\"99\", \"Google Chrome\";v=\"104\"")
		r.Header.Set("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36")
	})
	if err != nil {
		return nil, err
	}
	return result, nil
}

// https://www.tiktok.com/@ryanbakery/video/7162083541365312795?is_from_webapp=1&sender_device=pc&web_id=7146322967022372354
func IsTikTokUri(u string) bool {
	fmt.Println(u, strings.HasPrefix("https://www.tiktok.com", u))
	return strings.HasPrefix(u, "https://www.tiktok.com")
}

File: app/src/main/java/euphoria/psycho/fileserver/TikTok.java
package euphoria.psycho.fileserver;

import org.json.JSONObject;

import java.net.HttpURLConnection;
import java.net.InetSocketAddress;
import java.net.Proxy;
import java.net.Proxy.Type;
import java.net.URL;

import euphoria.psycho.fileserver.VideoDatabase.Video;

public class TikTok {
    public static Video fetch(String uri) throws Exception {
        HttpURLConnection c = (HttpURLConnection) new URL("https://www.tikwm.com/api/?count=12&cursor=0&web=1&hd=1&url=" + uri)
                .openConnection(new Proxy(Type.HTTP, new InetSocketAddress("127.0.0.1", 10809)));
        c.addRequestProperty("Accept", "application/json, text/javascript, */*; q=0.01");
        c.addRequestProperty("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8");
        c.addRequestProperty("sec-ch-ua", "\"Chromium\";v=\"104\", \" Not A;Brand\";v=\"99\", \"Google Chrome\";v=\"104\"");
        c.addRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36");
        JSONObject jsonObject = new JSONObject(Shared.readString(c));
        Video video = new Video();
        video.Title = jsonObject.getJSONObject("data").getString("title");
        video.Url = uri;
        video.Play = "https://www.tikwm.com/" + jsonObject.getJSONObject("data").getString("hdplay");
        video.MusicPlay = jsonObject.getJSONObject("data").getJSONObject("music_info").getString("play");
        video.MusicTitle = jsonObject.getJSONObject("data").getJSONObject("music_info").getString("title");
        video.MusicAuthor = jsonObject.getJSONObject("data").getJSONObject("music_info").getString("author");
        video.Cover = "https://www.tikwm.com/" + jsonObject.getJSONObject("data").getString("cover");
        video.CreateAt = System.currentTimeMillis();
        video.UpdateAt = System.currentTimeMillis();
        return video;
    }
}
